source/server/admin/logs_handler.cc

```diff
@@ -56,56 +60,96 @@ bool LogsHandler::changeLogLevel(const Http::Utility::QueryParams& params) {
     return false;
   }
 
-  std::string name = params.begin()->first;
-  std::string level = params.begin()->second;
+  const auto it = params.begin();
+  absl::string_view key = it->first;
+  absl::string_view value = it->secondn;
+
+  if (key == "paths") {
+    // Bulk change log level by name:level pairs
+    std::vector<absl::string_view> pairs = absl::StrSplit(value, ',', absl::SkipWhitespace());
+    bool ret = true;
+    for (const auto& name_value : pairs) {
+      std::pair<absl::string_view, absl::string_view> name_value_pair =
+          absl::StrSplit(name_value, absl::MaxSplits(':', 1), absl::SkipWhitespace());
+
+      auto name = name_value_pair.first;
+      auto value = name_value_pair.second;
+
+      if (name.empty() || value.empty()) {
+        continue;
+      }
+
+      ret = changeLogLevelByName(name, value) && ret;
+    }
 
-  // First see if the level is valid.
-  size_t level_to_use = std::numeric_limits<size_t>::max();
+    return ret;
+  } else if (key == "level") {
+    // Change all log levels
+    return changeAllLogLevels(value);
+  } else {
+    // Change specific log level by its name
+    return changeLogLevelByName(key, value);
+  }
+}
+
+static bool parseLevel(absl::string_view level, spdlog::level::level_enum& level_to_use) {
```

Comment:
nit: return absl::optional<spdlog::level::level_enum> for this pattern rather than returning a bool and modifying a reference param.